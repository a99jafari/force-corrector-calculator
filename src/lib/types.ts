export type Digit = "0" | "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9" | ".";
export type Operator = "+" | "*" | "-" | "/";
export type HighPriorityOperator = "*" | "/";
export type LowPriorityOperator = "+" | "-";

export type Key = Digit | Operator;

export const DIGITS: Digit[] = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "."];
export const OPERATORS: Operator[] = ["+", "*", "-", "/"];
export const DIGITS_AND_OPERATORS: (Digit | Operator)[] = [...DIGITS, ...OPERATORS];
export const CALCULATOR_VALID_INPUT_KEYS: (Digit | Operator | ".")[] = [...DIGITS_AND_OPERATORS, "."]
export const CALCULATOR_KEYS = [...DIGITS_AND_OPERATORS]

export const CALCULATOR_ACCEPTABLE_KEYS = [
  ...DIGITS_AND_OPERATORS,
  "Enter",
  "Delete"
];
