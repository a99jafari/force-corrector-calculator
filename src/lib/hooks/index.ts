export { useComputableString } from "./computable-string";
