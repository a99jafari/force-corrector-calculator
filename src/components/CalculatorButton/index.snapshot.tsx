import { render } from "@testing-library/react";
import { DIGITS_AND_OPERATORS } from "../../lib/types";
import { CalculatorButton } from "..";
import faker from "faker";


describe("CalculatorButton test", () => {
  faker.seed(1);

  test("SNapshot test", () => {
    const symbol = faker.random.arrayElement(DIGITS_AND_OPERATORS);
    const type = faker.random.arrayElement(["digit", "operator"]) as "digit" | "operator";
    
    const { container } = render(
      <CalculatorButton type={type} symbol={symbol} onClick={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });
});
