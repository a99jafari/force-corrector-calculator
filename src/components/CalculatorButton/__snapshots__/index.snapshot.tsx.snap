// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorButton test SNapshot test 1`] = `
<div>
  <button
    class="operator"
    value="6"
  >
    6
  </button>
</div>
`;
